### DISCUSSION TITLE
Isso in tricky network

### DISCUSSION THREAD

**Discussion by stalkerGH** [posativ/isso#1033] [Q&A]
Hi Isso users.

I wonder if it is at all possible to run Isso in such tricky configuration:

1. static site hosted on paid server (hosting company),
2. Isso self-hosted on Raspberry Pi (Raspbian OS) in my home LAN, behind a router.

Maybe someone owns similar configuration - not exactly on RPi but splitted on two different places/servers and networks - and could give some hints?

**Comment by jelmer** [comment]
This should indeed be possible, so long as you have a way of making the self-hosted isso behind the router publically accessible.

**Comment by stalkerGH** [comment]
Thank you for reply. What ports should I open on the router and Raspberry then?

I found several tutorials on the web which describe process of install and configuration. They differ in some points, for example in isso.conf - some use certain configuration keys, some not. So I'm stuck. For the future reference, in random order:

- https://therandombits.com/2018/12/how-to-add-isso-comments-to-your-site/
- https://stanislas.blog/2018/02/add-comments-to-your-blog-with-isso/
- https://blog.phusion.nl/2018/08/16/isso-simple-self-hosted-commenting-system/#1installingisso
- https://oktomus.com/posts/2020/add-comments-to-a-static-blog-with-isso/

**Reply by jelmer** [reply]
What ports you open would depend on your exact config. 

You do need to have your isso config listening on 0.0.0.0 rather than localhost or 127.0.0.1.

E.g. open port 8080, follow the instructions from the first article but set the listen_address to http://0.0.0.0:8080

**Reply by stalkerGH** [reply]
Thank you, I'll try then.

**Reply by jelmer** [reply]
Note that this won't get you https for your isso instance if you directly include your raspberry pi's IP/port in your static config. Most browsers will probably give you a security warning of some sort. 

The way around this would be to run a reverse http proxy for isso, either on your raspberry pi or elsewhere (maybe your static site host can do this?).

**Comment by stalkerGH** [comment]
OK. Partial success. I got Isso to work on my RPi and could reach /admin from the browser (RPi is in DMZ). It is available at http://stalker.hopto.org/admin/ But I can't get Nginx to work. My configs are:

```
server {
        listen 80;
        listen [::]:80;
        server_name stalker.hopto.org;
        return 301 http://stalker.hopto.org$request_uri;

        access_log /dev/null;
        error_log /dev/null;
}
```

```
[general]
dbpath = /var/lib/isso/comments.db
host = my_domain <--- here of course is real working TLD
max-age = 15m
notify = smtp
log-file = /var/log/isso.log

[moderation]
enabled = false

[admin]
enabled=true
password = ***

[server]
listen = http://0.0.0.0:8080
public-endpoint = http://stalker.hopto.org
reload = off
profile = off

[guard]
enabled = true
ratelimit = 2
direct-reply = 3
reply-to-self = false
require-author = false
require-email = false
```

Nginx log is empty all the time. Isso log contains:

Traceback (most recent call last):
  File "/home/isso/.local/lib/python3.7/site-packages/isso/ext/notifications.py", line 65, in __init__
    with SMTPConnection(self.conf):
  File "/home/isso/.local/lib/python3.7/site-packages/isso/ext/notifications.py", line 36, in __enter__
    timeout=self.conf.getint('timeout'))
  File "/usr/lib/python3.7/smtplib.py", line 251, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/lib/python3.7/smtplib.py", line 336, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/lib/python3.7/smtplib.py", line 307, in _get_socket
    self.source_address)
  File "/usr/lib/python3.7/socket.py", line 727, in create_connection
    raise err
  File "/usr/lib/python3.7/socket.py", line 716, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused
Using database at '/var/lib/isso/comments.db'
connected to my_domain

What should I check next?

PS. My RPi is behind a router and stalker.hopto.org is no-ip.com DDNS service.

**Reply by BBaoVanC** [reply]
Are you still having issues? The nginx file looks incomplete. All it says is to redirect the website to itself. You'll need a proxy_pass command and some other things to get it working. Maybe try this guide: https://docs.nginx.com/nginx/admin-guide/web-server/reverse-proxy/

The error you're seeing in the Isso log seems to be caused by a misconfigured mail server; if you aren't trying to use mail to send new comment notifications, try setting `notify` in the `[general]` section to stdout instead. I don't remember if this entirely disables the SMTP integration though.

**Reply by stalkerGH** [reply]
@BBaoVanC Thank you for reply but I gave up after several days. Maybe I should try again with more calm :)
